app/models/user.rb

```diff
@@ -7,4 +7,12 @@ class User < ActiveRecord::Base
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+  def can_edit_piece?(piece)
+    self.admin || self == piece.user
+  end
+
+  def can_edit_review?(review)
+    self.admin || self == review.user
```

Comment:
Ignoring the difference between `piece` and `review`, notice that `can_edit_piece?` and `can_edit_review?` have the exact same logic? It would be cleaner to implement them as the same method:

```
def can_edit?(resource)
  admin || self == resource.user
end
```

Notice you can skip the first `self` - the User instance is the receiver of this method, so it knows (by method lookup) that `admin` refers to `self.admin`, hence the 'redundant self detected' warning